core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -460,11 +475,10 @@ public void removeBlocks(List<Long> blockIds, boolean delete) throws Unavailable
             // Make sure blockId is removed from mLostBlocks when the block metadata is deleted.
             // Otherwise blockId in mLostBlock can be dangling index if the metadata is gone.
             mLostBlocks.remove(blockId);
-            if (mBlocks.remove(blockId) != null) {
-              JournalEntry entry = JournalEntry.newBuilder()
-                  .setDeleteBlock(DeleteBlockEntry.newBuilder().setBlockId(blockId)).build();
-              journalContext.append(entry);
-            }
+            mBlockStore.removeBlock(blockId);
```

Comment:
since this does not throw exception, it will journal the deletion even blockId does not exist. Do we need to consider on this case?

Reply:
our locking ensures that the block does exist at this point. We check `isPresent` after taking the lock:
```java
       try (LockResource lr = lockBlock(blockId)) {
          Optional<BlockMeta> block = mBlockStore.getBlock(blockId);
          if (!block.isPresent()) {
            continue;
          }
```